src/app/features/test-records/views/test-record/test-record.component.ts

```diff
@@ -17,60 +16,59 @@ import { BaseTestRecordComponent } from '../../components/base-test-record/base-
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class TestRecordComponent implements OnInit {
-  @ViewChild(BaseTestRecordComponent) private set baseTestRecordComponent(c: BaseTestRecordComponent) {
-    c?.dynamicFormGroupComponents?.forEach((component) => {
-      this.sectionForms.push(component.form);
-    });
+  @ViewChild(BaseTestRecordComponent) private set baseTestRecordComponent(component: BaseTestRecordComponent) {
+    component?.dynamicFormGroupComponents?.forEach(component => this.sectionForms.push(component.form));
   }
 
-  sectionForms: Array<CustomFormGroup | CustomFormArray> = [];
-
-  defectTpl: FormNode = DefectTpl;
+  isEditing$: Observable<boolean> = of(false);
   testResult$: Observable<TestResultModel | undefined> = of(undefined);
-  defectsData$: Observable<Defects | undefined> = of(undefined);
-  edit$: Observable<boolean> = of(false);
+  defects$: Observable<Defects | undefined> = of(undefined);
+
+  sectionForms: Array<CustomFormGroup | CustomFormArray> = [];
+  defectForms: Array<CustomFormGroup | CustomFormArray> = [];
 
   constructor(
-    private testRecordsService: TestRecordsService,
-    private routerService: RouterService,
-    private router: Router,
+    private errorService: GlobalErrorService,
     private route: ActivatedRoute,
-    private errorService: GlobalErrorService
+    private router: Router,
+    private routerService: RouterService,
+    private testRecordsService: TestRecordsService,
   ) {}
 
-  ngOnInit() {
+  ngOnInit(): void {
+    this.isEditing$ = this.routerService.routeEditable$;
     this.testResult$ = this.testRecordsService.testResult$;
-    this.defectsData$ = this.testRecordsService.defectData$;
-    this.edit$ = this.routerService.routeEditable$;
+    this.defects$ = this.testRecordsService.defectData$;
   }
 
-  handleEdit() {
+  defectFormsChange(forms: Array<CustomFormGroup | CustomFormArray>): void {
+    this.defectForms = forms;
+  }
+
+  handleEdit(): void {
     this.router.navigate([], { queryParams: { edit: true }, queryParamsHandling: 'merge', relativeTo: this.route });
   }
 
-  handleCancel() {
+  handleCancel(): void {
     this.router.navigate([], { queryParams: { edit: false }, queryParamsHandling: 'merge', relativeTo: this.route });
   }
 
-  get sectionFormsInvalid() {
-    return [...this.sectionForms].map((f) => f.invalid).some((b) => b);
-  }
+  async handleSave(): Promise<void> {
+    this.sectionForms.concat(this.defectForms.filter(f => f));
 
-  async handleSave() {
-    this.sectionForms.forEach((f) => {
-      const errors = DynamicFormService.updateValidity(f);
+    this.sectionForms.forEach(form => {
+      const errors = DynamicFormService.updateValidity(form);
       errors.length > 0 && this.errorService.patchErrors(errors);
     });
 
-    if (this.sectionFormsInvalid) {
+    if (this.sectionForms.some(form => form.invalid)) {
```

Comment:
I'm not sure whether I prefer to have `this.sectionFromsInvalid` in terms of readability as the name implies what it's doing. But perhaps tweaking the implementation of `this.sectionFromsInvalid` to:

```
get sectionFormsInvalid() {
    return this.sectionForms.some(form => form.invalid);
}
```

Reply:
I am with you about the readability and I would've definitely left this as a getter if it was being used in multiple places. But abstracting a single line of code that is only used once is redundant. Thus, I will leave this as is, but keep your suggestion in mind for future additions.